### ISSUE TITLE
Use of "_" to prefix OSHMEM constant names

### ISSUE THREAD

**Issue by jsquyres** [openshmem-org/tests-uh#17]
I notice that to get these tests to compile in Open MPI's OpenSHMEM implementation, I had to make the following changes:

```patch
diff --git a/feature_tests/C/test_shmem_query.c b/feature_tests/C/test_shmem_que
index 3c54de2..a611240 100644
--- a/feature_tests/C/test_shmem_query.c
+++ b/feature_tests/C/test_shmem_query.c
@@ -65,7 +65,7 @@
 int
 main ()
 {
-    char name[_SHMEM_MAX_NAME_LEN];
+    char name[SHMEM_MAX_NAME_LEN];
     int major_ver, minor_ver;
     int me;
 
@@ -79,7 +79,7 @@ main ()
       shmem_info_get_version (&major_ver, &minor_ver);
       shmem_info_get_name (name);
 
-      if (major_ver == _SHMEM_MAJOR_VERSION) {
+      if (major_ver == SHMEM_MAJOR_VERSION) {
           PRINT_VER("Major","Passed");
       }
       else {
@@ -87,7 +87,7 @@ main ()
           fail_count++;
       }
 
-      if (minor_ver == _SHMEM_MINOR_VERSION) {
+      if (minor_ver == SHMEM_MINOR_VERSION) {
           PRINT_VER("Minor","Passed");
       }
       else {
@@ -95,7 +95,7 @@ main ()
           fail_count++;
       }
 
-      if (strlen (name) < _SHMEM_MAX_NAME_LEN) {
+      if (strlen (name) < SHMEM_MAX_NAME_LEN) {
           PRINT_NAME("Passed");
       }
       else {
diff --git a/feature_tests/C/test_shmem_synchronization.c b/feature_tests/C/test
index df47afb..e2e9a1f 100644
--- a/feature_tests/C/test_shmem_synchronization.c
+++ b/feature_tests/C/test_shmem_synchronization.c
@@ -125,7 +125,7 @@ main (void)
         shmem_barrier_all ();
 
         if (me == 1) {
-            shmem_long_wait_until (dest, _SHMEM_CMP_NE, 9L);
+            shmem_long_wait_until (dest, SHMEM_CMP_NE, 9L);
             printf ("Test for explicit conditional wait: Passed\n");
         }
 
```

I'm not an OpenSHMEM expert.  Is Open MPI incorrect in not having the "_"-prefixed versions of these names?  Or are these tests incorrect?

Thanks!

**Comment by jdinan** [comment]
The ```_SHMEM_*``` constants were deprecated, but have not been removed, so implementations should still support them.  The deprecation documentation in the spec is admittedly pretty bad, but it will be much better starting with the next version of the OpenSHMEM spec.  Here's a draft that lists all of the deprecated constants: [main_spec.pdf](https://github.com/openshmem-org/tests-uh/files/711475/main_spec.pdf)

**Comment by jsquyres** [comment]
Gothca.

Do the non-"_"-prefixed versions of these constants exist in the spec?  (i.e., do I just need to rename the OMPI constants, or do I need to add the deprecated names)

**Comment by jdinan** [comment]
The older constants have both ```SHMEM_CONST``` and ```_SHMEM_CONST``` versions.  There are some new constants like ```SHMEM_SYNC_SIZE``` that don't have an ```_SHMEM_*``` equivalent.  This should be covered in the deprecation section of the 1.4 spec draft I posted in my previous comment, but you can cross reference OpenSHMEM 1.2 to be sure ([OpenSHMEM-1.2.pdf](https://github.com/openshmem-org/tests-uh/files/711701/OpenSHMEM-1.2.pdf)).

**Comment by jsquyres** [comment]
Got it; thanks.  To be addressed in open-mpi/ompi#2745.
